api/src/main/java/org/geysermc/geyser/api/item/custom/CustomItemData.java

```diff
@@ -114,6 +115,92 @@ public interface CustomItemData {
      */
     @NonNull Set<String> tags();
 
+    /**
+     * Gets the stack size of the item.
+     *
+     * Returns 0 if not set. When not set (or 0), takes the Java item stack count when based of a vanilla item, or uses 64 when porting a modded item.
```

Comment:
what's preventing us from inheriting the modded item's stack size?

Reply:
When Geyser registers the modded item mappings (in `CustomItemRegistryPopulator#registerCustomItem`), it doesn't actually have access to the properties of the (modded) Java item - all it knows is the properties specified in the mapping, and it creates a Java item instance from that (can be seen [here](https://github.com/GeyserMC/Geyser/blob/66f30a2cb6f477ec8ff9d9aa34ec015a6cd6891d/core/src/main/java/org/geysermc/geyser/registry/populator/CustomItemRegistryPopulator.java#L129C9-L159C91)). The current behaviour of modded item mappings is to default to 64 when no stack size is specified, which I copied in this PR.

Note that Geyser does have access to vanilla Java items, which is why it's able to copy the properties of those when creating a vanilla item mapping ([here](https://github.com/GeyserMC/Geyser/blob/66f30a2cb6f477ec8ff9d9aa34ec015a6cd6891d/core/src/main/java/org/geysermc/geyser/registry/populator/CustomItemRegistryPopulator.java#L102-L107)).

I'm not aware of a way for Geyser to automatically fetch the properties of a modded item - if there is, I could implement it in this PR.

Reply:
You're right; it doesn't look like it's currently possible to add a custom model data override to a modded item, so no need to inherit anything :p
Let's just rephrase it that 64 is the default for nonvanilla items, otherwise it inherits the default of the Java item